CommonPackage.cmake

```diff
@@ -14,44 +25,181 @@ set(ENV{PKG_CONFIG_PATH}
-  if(COMMON_PACKAGE_REQUIRED_POS EQUAL -1) # Optional find
-    find_package(${Name} ${COMMON_PACKAGE_FIND_QUIET} ${COMMON_PACKAGE_ARGS}) # try standard cmake way
-    if((NOT ${Name}_FOUND) AND (NOT ${COMMON_PACKAGE_NAME}_FOUND) AND PKG_CONFIG_EXECUTABLE)
-      pkg_check_modules(${Name} ${Name}${COMMON_PACKAGE_VERSION}
-        ${COMMON_PACKAGE_FIND_QUIET}) # try pkg_config way
+  if(EXISTS ${PROJECT_SOURCE_DIR}/CMake/FindPackagesPost.cmake)
+    include(${PROJECT_SOURCE_DIR}/CMake/FindPackagesPost.cmake)
+  endif()
+
+  string(TOUPPER ${PROJECT_NAME} UPPER_PROJECT_NAME)
+  set(DEFDEP "${UPPER_PROJECT_NAME}_USE_${COMMON_PACKAGE_NAME}")
+  string(REGEX REPLACE "-" "_" DEFDEP ${DEFDEP})
+
+  if(${COMMON_PACKAGE_NAME}_FOUND)
+    set(${Name}_name ${COMMON_PACKAGE_NAME})
+    set(${Name}_FOUND TRUE)
+  elseif(${Name}_FOUND)
+    set(${Name}_name ${Name})
+    set(${COMMON_PACKAGE_NAME}_FOUND TRUE)
+  else()
+    set(${PROJECT_NAME}_FIND_PACKAGES_NOTFOUND "${${PROJECT_NAME}_FIND_PACKAGES_NOTFOUND} ${Name}")
+  endif()
+  if(${Name}_name)
+    list(APPEND FIND_PACKAGES_DEFINES ${DEFDEP})
+    if(NOT COMMON_LIBRARY_TYPE MATCHES "SHARED")
+      list(APPEND ${UPPER_PROJECT_NAME}_DEPENDENT_LIBRARIES ${Name})
+    endif()
+    set(${PROJECT_NAME}_FIND_PACKAGES_FOUND "${${PROJECT_NAME}_FIND_PACKAGES_FOUND} ${Name}")
+    link_directories(${${${Name}_name}_LIBRARY_DIRS})
+    if(NOT "${${${Name}_name}_INCLUDE_DIRS}" MATCHES "-NOTFOUND")
+      if(COMMON_PACKAGE_SYSTEM)
+        include_directories(BEFORE SYSTEM ${${${Name}_name}_INCLUDE_DIRS})
+      else()
+        include_directories(${${${Name}_name}_INCLUDE_DIRS})
+      endif()
     endif()
-  else() # required find
-    list(REMOVE_AT COMMON_PACKAGE_ARGS ${COMMON_PACKAGE_REQUIRED_POS})
-    find_package(${Name} ${COMMON_PACKAGE_FIND_QUIET} ${COMMON_PACKAGE_ARGS}) # try standard cmake way
-    if((NOT ${Name}_FOUND) AND (NOT ${COMMON_PACKAGE_NAME}_FOUND) AND PKG_CONFIG_EXECUTABLE)
-      pkg_check_modules(${Name} REQUIRED ${Name}${COMMON_PACKAGE_VERSION}
-        ${COMMON_PACKAGE_FIND_QUIET}) # try pkg_config way (and fail if needed)
+    if(NOT "${${${Name}_name}_INCLUDE_DIR}" MATCHES "-NOTFOUND")
+      if(COMMON_PACKAGE_SYSTEM)
+        include_directories(BEFORE SYSTEM ${${${Name}_name}_INCLUDE_DIR})
+      else()
+        include_directories(${${${Name}_name}_INCLUDE_DIR})
+      endif()
     endif()
   endif()
 endmacro()
+
+
+macro(common_package_post)
+  # Write defines.h and options.cmake
+  if(NOT PROJECT_INCLUDE_NAME)
+    message(FATAL_ERROR "PROJECT_INCLUDE_NAME not set, old or missing Common.cmake?")
+  endif()
+  if(NOT OPTIONS_CMAKE)
+    set(OPTIONS_CMAKE ${CMAKE_CURRENT_BINARY_DIR}/options.cmake)
+  endif()
+
+  configure_file(${CMAKE_SOURCE_DIR}/CMake/common/cpp/defines.h
+    ${OUTPUT_INCLUDE_DIR}/${PROJECT_INCLUDE_NAME}/defines.h @ONLY)
+  set(DEFINES_FILE "${CMAKE_CURRENT_BINARY_DIR}/include/${PROJECT_INCLUDE_NAME}/defines${SYSTEM}.h")
+  list(APPEND COMMON_INCLUDES
```

Comment:
Is COMMON_INCLUDES still used somewhere? If yes, it must be documented in the output variables. If not, then defines.h does not get installed anymore, despite the fact that api.h includes it.